test/test_convergence.jl

```diff
@@ -103,111 +35,76 @@ function input_config(ArrayType; n = 10, n_RS1 = 20, n_RS2 = 20, T_surface = 290
     return z_range, relative_sat, T_surface, T_min
 end
 
-"""
-    shared_profiles(
-        param_set::APS,
-        z_range::AbstractArray,
-        relative_sat::AbstractArray,
-        T_surface,
-        T_min,
-    )
-
-Compute profiles shared across `PhaseDry`,
-`PhaseEquil` and `PhaseNonEquil` thermodynamic
-states, including:
- - `z` altitude
- - `T` temperature
- - `p` pressure
- - `RS` relative saturation
-"""
-function shared_profiles(param_set::APS, z_range::AbstractArray, relative_sat::AbstractArray, T_surface, T_min)
-    FT = eltype(z_range)
-    n_RS = length(relative_sat)
-    n = length(z_range)
-    T = similar(z_range, n * n_RS)
-    p = similar(z_range, n * n_RS)
-    RS = similar(z_range, n * n_RS)
-    z = similar(z_range, n * n_RS)
-    linear_indices = LinearIndices((1:n, 1:n_RS))
-    # We take the virtual temperature, returned here,
-    # as the temperature, and then compute a thermodynamic
-    # state consistent with that temperature. This profile
-    # will not be in hydrostatic balance, but this does not
-    # matter for the thermodynamic test profiles.
-    profile = TP.DecayingTemperatureProfile{FT}(param_set, FT(T_surface), FT(T_min))
-    for i in linear_indices.indices[1]
-        for j in linear_indices.indices[2]
-            k = linear_indices[i, j]
-            z[k] = z_range[i]
-            T[k], p[k] = profile(param_set, z[k])
-            RS[k] = relative_sat[j]
-        end
-    end
-    return z, T, p, RS
-end
-
-profiles_sfc = []
-profiles_int = []
-profiles = PhaseEquilProfiles(thermo_params, Array{FT});
-
-## Properties contained in `profiles`
-## :z, :T, :p, :RS, :e_int, :h, :ρ, :θ_liq_ice, :q_tot, :q_liq, :q_ice, :q_pt, :RH, :e_pot, :u, :v, :w, :e_kin, :phase_type
-
-for prof in profiles
-    if prof.z == FT(0)
-        push!(profiles_sfc, prof)
-    else
-        push!(profiles_int, prof)
-    end
+function generate_profiles(FT)
+    toml_dict = CP.create_toml_dict(FT; dict_type = "alias")
+    param_set = create_parameters(toml_dict, UF.BusingerType())
+    thermo_params = SFP.thermodynamics_params(param_set)
+    uft = SFP.universal_func_type(param_set)
+    profiles = collect(TD.TestedProfiles.PhaseEquilProfiles(thermo_params, Array{FT}))
+    profiles_sfc = filter(p->iszero(p.z), profiles)
```

Comment:
@charleskawczynski `push`ing to generic container swapped out for `filter`ed collection.